e2e/tests/desktop/guest-shopper.spec.js

```diff
@@ -0,0 +1,129 @@
+    page.getByRole("heading", { name: /Contact Info/i })
+  ).toBeVisible();
+
+  await page.locator("input#email").fill("test@gmail.com");
+
+  await page.getByRole("button", { name: /Checkout as guest/i }).click();
+
+  // Confirm the email input toggles to show edit button on clicking "Checkout as guest"
+  const step0Card = page.locator("div[data-testid='sf-toggle-card-step-0']");
+
+  await expect(step0Card.getByRole("button", { name: /Edit/i })).toBeVisible();
+
+  await expect(
+    page.getByRole("heading", { name: /Shipping Address/i })
+  ).toBeVisible();
+
+  await page.locator("input#firstName").fill("John");
+  await page.locator("input#lastName").fill("Doe");
+  await page.locator("input#phone").fill("8572068547");
+  await page.locator("input#address1").fill("5 Wall St.");
+  await page.locator("input#city").fill("Burlington");
+  await page.locator("select#stateCode").selectOption("MA");
+  await page.locator("input#postalCode").fill("01803");
+
+  await page
+    .getByRole("button", { name: /Continue to Shipping Method/i })
+    .click();
+
+  // Confirm the shipping details form toggles to show edit button on clicking "Checkout as guest"
+  const step1Card = page.locator("div[data-testid='sf-toggle-card-step-1']");
+
+  await expect(step1Card.getByRole("button", { name: /Edit/i })).toBeVisible();
+
+  await expect(
+    page.getByRole("heading", { name: /Shipping & Gift Options/i })
+  ).toBeVisible();
+
+  await page.getByRole("button", { name: /Continue to Payment/i }).click();
+
+  // Confirm the shipping options form toggles to show edit button on clicking "Checkout as guest"
+  const step2Card = page.locator("div[data-testid='sf-toggle-card-step-2']");
+
+  await expect(step2Card.getByRole("button", { name: /Edit/i })).toBeVisible();
+
+  await expect(page.getByRole("heading", { name: /Payment/i })).toBeVisible();
+
+  const creditCardExpiry = getCreditCardExpiry();
+
+  await page.locator("input#number").fill("4111111111111111");
+  await page.locator("input#holder").fill("John Doe");
+  await page.locator("input#expiry").fill(creditCardExpiry);
+  await page.locator("input#securityCode").fill("213");
+
+  await page.getByRole("button", { name: /Review Order/i }).click();
+
+  // Confirm the shipping options form toggles to show edit button on clicking "Checkout as guest"
+  const step3Card = page.locator("div[data-testid='sf-toggle-card-step-3']");
+
+  await expect(step3Card.getByRole("button", { name: /Edit/i })).toBeVisible();
+  page
+    .getByRole("button", { name: /Place Order/i })
+    .first()
+    .click();
+
+  const orderConfirmationHeading = page.getByRole("heading", {
+    name: /Thank you for your order!/i,
+  });
+  await orderConfirmationHeading.waitFor();
+
+  await expect(
+    page.getByRole("heading", { name: /Order Summary/i })
+  ).toBeVisible();
+  await expect(page.getByText(/2 Items/i)).toBeVisible();
+  await expect(page.getByRole("link", { name: /Stripe Shell/i })).toBeVisible();
```

Comment:
Not urgent but could we add some additional checks to verify other properties are being set on the order on checkout? (the missing order email issue is still fresh for me)

Reply:
We can have a follow up story for adding more test-cases. We need to go over having stricter checks for generated apps, config, more e2e test flows for the deployed app. We can add this as a part of that work.